src/features/view-nft/Attributes/Attributes.tsx

```diff
@@ -0,0 +1,173 @@
+
+	const { data: metadata } = useDomainMetadata(domainId);
+
+	const [isToggled, setIsToggled] = useState<boolean>(true);
+	const [containerWidth, setContainerWidth] = useState<number>(0);
+
+	const initialVisibleAttributesCount: number = useMemo(() => {
+		const isTablet = containerWidth > 592 && containerWidth < 800;
+		const isMobile = containerWidth <= 592;
+
+		if (isMobile) return NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.MOBILE;
+		if (isTablet) return NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.TABLET;
+		return NFT_ATTRIBUTES_VISIBLE_COUNTS_BY_VIEWPORT.DESKTOP;
+	}, [containerWidth]);
+
+	const initialHiddenAttributesCount: number = Math.max(
+		metadata?.attributes?.length - initialVisibleAttributesCount,
+		0,
+	);
+
+	const visibleAttributes: Attribute[] = useMemo(() => {
+		if (!metadata?.attributes) {
+			return [];
+		}
+
+		const visibleAttributesCount = isToggled
+			? initialVisibleAttributesCount
+			: metadata.attributes.length;
+
+		return metadata.attributes.slice(0, visibleAttributesCount);
+	}, [metadata, isToggled, initialVisibleAttributesCount]);
+
+	if (visibleAttributes.length === 0) {
+		return null;
+	}
+
+	useResize({
+		onResize: setContainerWidth,
+		targetRef: containerRef,
+	});
+
+	return (
+		<section className={styles.AttributesSection}>
+			<div className={styles.Container} ref={containerRef}>
+				<h4>Attributes</h4>
+
+				<AttributesGrid
+					attributes={visibleAttributes}
+					isToggled={isToggled}
+					setIsToggled={setIsToggled}
+					initialHiddenAttributesCount={initialHiddenAttributesCount}
+				/>
+			</div>
+		</section>
+	);
+};
+
+/*******************
+ * Attributes Grid
+ *******************/
+
+interface AttributesGridProps {
+	attributes: Attribute[];
+	isToggled: boolean;
+	setIsToggled: (value: boolean) => void;
+	initialHiddenAttributesCount: number;
+}
+
+const AttributesGrid = ({
+	attributes,
+	isToggled,
+	setIsToggled,
+	initialHiddenAttributesCount,
+}: AttributesGridProps) => {
+	return (
+		<ul className={styles.Grid}>
+			{attributes.map((attributeItem: Attribute, index: number) => (
+				<AttributeItem
+					key={index}
+					attributeItem={attributeItem}
+					attributeIndex={index}
+				/>
+			))}
+
+			{initialHiddenAttributesCount > 0 && (
+				<ToggleButton
+					count={initialHiddenAttributesCount}
+					isToggled={isToggled}
+					setIsToggled={setIsToggled}
+				/>
+			)}
+		</ul>
+	);
+};
+
+/*******************
+ * Attribute Item
+ *******************/
+
+interface AttributeItemProps {
+	attributeItem: Attribute;
+	attributeIndex: number;
+}
+
+const AttributeItem = ({
+	attributeItem,
+	attributeIndex,
+}: AttributeItemProps) => {
+	return (
+		<li
+			className={cx(styles.AttributeItem, {
+				setOpacityAnimation: attributeIndex > 10,
+			})}
+			key={`attribute-type-${attributeItem.trait_type}`}
+		>
+			<span className={styles.Traits}>{attributeItem.trait_type}</span>
+			<span className={styles.Properties}>{attributeItem.value} </span>
+		</li>
+	);
+};
```

Comment:
nit: components are a little more flexible if you pass in specific values of an object, rather than the whole object, e.g.
```
const AttributeItem = ({
	attributeType,
    attributeValue,
	attributeIndex,
}: AttributeItemProps) => {
	...
};
```
But not a big issue here as this is just a small subcomponent.

It could also be a good idea to move the `attributeIndex > 10` logic out of this component, and instead pass in the `setOpacityAnimation` class name when the index in `map` is `> 10`. Again, not a big issue, but a good principle to follow in general I think.

Reply:
Passed specific values at first, not sure why I decided to do the whole object in the end but makes sense thanks mate. Updated!